docs/loading/Flink-connector-starrocks.md

```diff
@@ -1,173 +1,647 @@
+    RowDataTransformer transformer = new RowDataTransformer();
+    // Create the sink with schema, options, and transformer
+    SinkFunction<RowData> starRockSink = StarRocksSink.sink(schema, options, transformer);
+    source.addSink(starRockSink);
+    ```
+
+  - The `RowDataTransformer` in the main program is defined as
+
+    ```java
+    private static class RowDataTransformer implements StarRocksSinkRowBuilder<RowData> {
+    
+        /**
+         * Set each element of the object array according to the input RowData.
+         * The schema of the array matches that of StarRocks table.
+         */
+        @Override
+        public void accept(Object[] internalRow, RowData rowData) {
+            internalRow[0] = rowData.id;
+            internalRow[1] = rowData.name;
+            internalRow[2] = rowData.score;
+            // Only need for StarRocks primary key table. Set the last
+            // element to tell whether the record is a UPSERT or DELETE.
+            internalRow[internalRow.length - 1] = StarRocksSinkOP.UPSERT.ordinal();
+        }
+    }  
+    ```
+
+## Best Practices
+
+### Load data to primary key table
+
+This section will show how to load data to StarRocks primary key table to achieve partial update, and conditional update.
+You can see [Change data through loading](https://docs.starrocks.io/en-us/latest/loading/Load_to_Primary_Key_tables) for the introduction of those features.
+These examples use Flink SQL.
+
+#### Preparations
+
+Create a database `test` and create a Primary Key table `score_board` in StarRocks.
+
+```SQL
+CREATE DATABASE `test`;
+
+CREATE TABLE `test`.`score_board`
+(
+    `id` int(11) NOT NULL COMMENT "",
+    `name` varchar(65533) NULL DEFAULT "" COMMENT "",
+    `score` int(11) NOT NULL DEFAULT "0" COMMENT ""
+)
+ENGINE=OLAP
+PRIMARY KEY(`id`)
+COMMENT "OLAP"
+DISTRIBUTED BY HASH(`id`);
+```
+
+#### Partial update
+
+This example will show how to load data only to columns `id` and `name`.
+
+1. Insert initial data to StarRocks table in MySQL client
+
+    ```SQL
+    mysql> INSERT INTO `score_board` VALUES (1, 'starrocks', 100), (2, 'flink', 100);
+
+    mysql> select * from score_board;
+    +------+-----------+-------+
+    | id   | name      | score |
+    +------+-----------+-------+
+    |    1 | starrocks |   100 |
+    |    2 | flink     |   100 |
+    +------+-----------+-------+
+    2 rows in set (0.02 sec)
+    ```
+
+2. Create a Flink table `score_board` in Flink SQL client
+
+   - Define the DDL which only includes columns `id` and `name`
+   - Set the option `sink.properties.partial_update` to `true` which tells the connector to do partial update.
+   - If the connector version <= 1.2.7, also need to set the option `sink.properties.columns` to `id,name,__op` which tells the connector the columns to update. Note you need append `__op` at the end, and this field is used to specify UPSERT/DELETE operation, but its value is set by the connector automatically.
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/cdc120a0-ced4-48b1-984f-9d4a1f80f3e0)
"who" need to???